### ISSUE TITLE
Uncaught RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by panins** [sveale/remote-edit#126]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.11
**System**: Mac OS X 10.10.5
**Thrown From**: [remote-edit](https://github.com/sveale/remote-edit) package, v1.8.18

### Stack Trace

Uncaught RangeError: Maximum call stack size exceeded

```
At /Users/nKt/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:185

RangeError: Maximum call stack size exceeded
    at Function.keys (native)
    at SFTPStream._cleanup (/Users/nKt/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:185:21)
    at SFTPStream.end (/Users/nKt/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:160:8)
    at SFTPWrapper.end (/Users/nKt/.atom/packages/remote-edit/node_modules/ssh2/lib/SFTPWrapper.js:29:23)
    at /Users/nKt/.atom/packages/remote-edit/lib/model/sftp-host.coffee:155:14
    at /Users/nKt/.atom/packages/remote-edit/lib/model/sftp-host.coffee:153:76
    at onerror (/Users/nKt/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:1027:7)
    at /Users/nKt/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:1072:30
    at SFTPStream._cleanup (/Users/nKt/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:190:38)
    at SFTPStream.end (/Users/nKt/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:160:8)
```

### Commands

```
     -0:07.3.0 remote-edit:browse (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ],
    "disabledPackages": [
      "language-apache"
    ]
  },
  "remote-edit": {
    "foldersOnTop": true
  }
}
```

### Installed Packages

```coffee
# User
Remote-FTP, v0.7.2
language-ini, v1.12.0
remote-edit, v1.8.18

# Dev
No dev packages
```

**Comment by ahollenbach** [comment]
I had to downgrade to 1.8.14 to solve this issue (temporarily).

**Comment by maurinator** [comment]
^ 1.8.14 works for me as well! thanks for the comment on downgrading!

**Comment by ruvasik** [comment]
I have last versions atom && remote-edit and view this error

RangeError: Maximum call stack size exceeded
    at SFTPStream.end (/Users/ruvasik/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js)
    at SFTPWrapper.end (/Users/ruvasik/.atom/packages/remote-edit/node_modules/ssh2/lib/SFTPWrapper.js:29:23)
    at /Users/ruvasik/.atom/packages/remote-edit/lib/model/sftp-host.coffee:155:14
    at /Users/ruvasik/.atom/packages/remote-edit/lib/model/sftp-host.coffee:153:76
    at onerror (/Users/ruvasik/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:1027:7)
    at /Users/ruvasik/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:1072:30
    at SFTPStream._cleanup (/Users/ruvasik/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:190:38)
    at SFTPStream.end (/Users/ruvasik/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:160:8)
    at SFTPWrapper.end (/Users/ruvasik/.atom/packages/remote-edit/node_modules/ssh2/lib/SFTPWrapper.js:29:23)
    at /Users/ruvasik/.atom/packages/remote-edit/lib/model/sftp-host.coffee:155:14

**Comment by markusdanek** [comment]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.11
**System**: Mac OS X 10.10.5
**Thrown From**: [remote-edit](https://github.com/sveale/remote-edit) package, v1.8.18


### Stack Trace

Uncaught RangeError: Maximum call stack size exceeded

```
At /Users/mdanek/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:185

RangeError: Maximum call stack size exceeded
    at Function.keys (native)
    at SFTPStream._cleanup (/Users/mdanek/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:185:21)
    at SFTPStream.end (/Users/mdanek/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:160:8)
    at SFTPWrapper.end (/Users/mdanek/.atom/packages/remote-edit/node_modules/ssh2/lib/SFTPWrapper.js:29:23)
    at /Users/mdanek/.atom/packages/remote-edit/lib/model/sftp-host.coffee:155:14
    at /Users/mdanek/.atom/packages/remote-edit/lib/model/sftp-host.coffee:153:76
    at onerror (/Users/mdanek/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:1027:7)
    at /Users/mdanek/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:1072:30
    at SFTPStream._cleanup (/Users/mdanek/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:190:38)
    at SFTPStream.end (/Users/mdanek/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/sftp.js:160:8)
```

### Commands

```
     -0:22.4.0 remote-edit:browse (atom-text-editor.editor.is-focused)
     -0:09.5.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:07.1.0 core:move-down (atom-text-editor.editor.mini.is-focused)
  2x -0:06.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:02.4.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:01.0 core:confirm (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "projectHome": "/Users/mdanek/Development/",
    "ignoredNames": [
      ".DS_Store",
      ".css.map",
      ".map",
      ".git",
      ".sass-cache"
    ],
    "disabledPackages": [
      "wrap-guide"
    ]
  },
  "remote-edit": {
    "clearFileList": true,
    "rememberLastOpenDirectory": true,
    "foldersOnTop": true
  }
}
```

### Installed Packages

```coffee
# User
language-typoscript2, v0.2.0
minimap, v4.13.3
remote-edit, v1.8.18
tree-view-open-files, v0.2.4
typo3-fluid-atom-snippets, v1.2.0

# Dev
No dev packages
```

**Comment by jacobmischka** [comment]
Also having this problem on Linux x64

**Comment by branderjeremy** [comment]
Any update? I'm also having this issue.

**Comment by lindgrenj6** [comment]
Reproduced the issue as well when just trying to edit a JSP file at the webserver's root.

**Comment by sveale** [comment]
Published a new version, 1.8.19, which should resolve the issue. Tested it on mac os x and win7.
